lib/extensions/descendant_loader.rb

```diff
@@ -165,8 +165,14 @@ def cache_path
 
     def load_cache
       return unless cache_path.exist?
-      data = YAML.load_file(cache_path)
+
+      data = File.open(cache_path, "r") do |f|
+        f.flock(File::LOCK_SH)
+        YAML.load(f.read)
```

Comment:
I think the cop may be right here
`s/load/load_file/`

Reply:
Don't think we want `load_file` we already have the file open and have a file handle